### DISCUSSION TITLE
Not sure why, but monitor doesn't show arrows (1.19.2)

### DISCUSSION THREAD

**Discussion by TylerWDoesStuff** [cc-tweaked/CC-Tweaked#1884] [Q&A] [answered]
Trying to integrate CC: Tweaked monitors into my Create lift, and I've gotten to the point where the code works but doesn't display correctly on the monitor. My code is below, as well as screenshots and a recording. It should show the lift's current floor letter with an arrow, but it only shows the floor letter.
```
local monitor = peripheral.find('monitor')
local elevFloor = peripheral.find('create_target')
monitor.clear()
monitor.setTextScale(3.5)
while true do
    if redstone.getInput('left') == true then
        print('here')
        monitor.setBackgroundColor(colors.white)
        monitor.setTextColor(colors.black)
    else
        print('herent') 
        monitor.setBackgroundColor(colors.black)
        monitor.setTextColor(colors.white)
    end
    monitor.clearLine()
    monitor.setCursorPos(1,1)
    local floorString = elevFloor.getLine(1)
    -- print(elevFloor.getLine(2))
    local liftPosTemp = string.gsub(elevFloor.getLine(2), "%%", "")
    local liftPos = tonumber(liftPosTemp)
    print(liftPos)
    print(floorString)
    if liftPos >= 9 and liftPos <= 50 then
        print('we gottem') -- this is a placeholder test, didn't know what wasn't working
        if string.match(floorString, "M") then
            monitor.write(floorString)
            print('going down')
            print(string.char(25)) 
            monitor.write(string.char(25))
        elseif string.match(floorString, "S") then
            monitor.write(floorString)
            print('going up')
            monitor.write(string.char(24))
        end
    else monitor.write(floorString)
    end
    sleep(1)
end
```
Recording (not sure why it was stuttering with OBS open): 

https://github.com/cc-tweaked/CC-Tweaked/assets/70175654/633dd79d-8a30-4939-a120-9a3f2b55db67

![2024-07-01_02 04 05](https://github.com/cc-tweaked/CC-Tweaked/assets/70175654/b139c0a8-cf5f-459f-8027-fd0f13597f62)
![2024-07-01_02 04 18](https://github.com/cc-tweaked/CC-Tweaked/assets/70175654/c7029044-0efa-4cb6-ba72-a9de3a20be9f)
I have a Threshold Switch attached to the Elevator Pulley, which tells me the position along with the Elevator / Redstone Contact.

**Comment by SquidDev** [comment, marked as answer]
I'm afraid I'm not super familiar with all the Create peripherals, but from what I can tell from [the implementation of `getLine`][impl], it returns a string which is always 32 characters long. This means when you write it to the monitor, the cursor is pushed off-screen to the right. This means the arrow won't actually be written to the screen!

The easiest fix here is either to truncate the `floorString` to be less than the monitor's width:

```lua
local width, height = monitor.getSize()
local floorString = elevFloor.getLine(1):sub(1, width - 1)
```

Or set the cursor position when writing the arrow:

```lua
local width, height = monitor.getSize()
monitor.setCursorPos(1, width)
monitor.write(string.char(24))
```

[impl]: https://github.com/tweaked-programs/cccbridge/blob/be3a8befc9630cb25d432b8fc7580f892e204fd5/forge/src/main/java/cc/tweaked_programs/cccbridge/common/computercraft/peripherals/TargetBlockPeripheral.java

**Reply by TylerWDoesStuff** [reply]
...huh. I'll try that and see if it helps, Kinda thought it would only draw as many chars as needed but I guess not?

**Reply by TylerWDoesStuff** [reply]
Works as expected now. Thanks!
